### PR TITLE
V0.2

### PR DESCRIPTION
**Pull request by ArthurJiang**
# Description

- Simulation toolkit
  - VM allocation environment
  - Dashboard for CIM and CitiBike
  - EventBuffer refine
  - Dump for snapshots, events and business engine special data.
- RL toolkit
  - New RL abstraction
  - More RL algorithms 
- Distributed toolkit
  - maro cli refactoring
  - maro cli support windows 
  - maro processes mode
  - maro grass on-premises mode
- Solution example
  - CIM: dqn & gnn example updated
  - Citi Bike: online LP example added
  - VM Scheduling: random & best fit example added

## Linked issue(s)/Pull request(s)

- [MARO processes mode doc](https://github.com/microsoft/maro/pull/230)
- [MARO on-premised mode](https://github.com/microsoft/maro/pull/220)
- [Learning model refactoring](https://github.com/microsoft/maro/pull/236)
- [MARO vis](https://github.com/microsoft/maro/pull/233)
- [MARO vis dump feature](https://github.com/microsoft/maro/pull/190)
- [RL toolkit doc update](https://github.com/microsoft/maro/pull/235)
- [VM scheduling doc](https://github.com/microsoft/maro/pull/228)
- [VM scheduling environment](https://github.com/microsoft/maro/pull/189)
- [VM data pipeline](https://github.com/microsoft/maro/pull/199)
- [EventBuffer refine](https://github.com/microsoft/maro/pull/197)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [x] New feature
- [x] Test
- [x] Doc update
- [x] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [x] Windows
  - [x] Mac OS
  - [x] Linux
- Python version:
  - [x] 3.6
  - [x] 3.7
- Key information snapshot(s):
  
## Needs Follow Up Actions

- [x] New release package
- [x] New docker image

## Checklist

- [x] Add/update the related comments
- [x] Add/update the related tests
- [x] Add/update the related documentations
- [x] Update the dependent downstream modules usage

### CODE REVIEW

File: CITATION [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@misc{MARO_MSRA
+title = {MARO: A Multi-Agent Resource Optimization Platform},
+author = {Arthur Jiang, Jia Zhang, Pingchao Yu, Lyuchun Huang, Yang Qiu, Jinyu Wang, Wenlei Shi, Kaiqi Li, Zhanyu Wang, Chengruidong Zhang, Tianyi Sun, Miaoran Chen, Kuanwei Yu, Xinran Wei, Michael Li, Ning Shang, Qiwei Meng, Shan Li, Jiang Bian, Biao Cheng, Tie-Yan Liu},
+year = {2020},
+publisher = {GitHub},
+journal = {GitHub repository},
+url = {https://github.com/microsoft/maro}
+}
```
Comment (Jinyu-W) [inline CITATION:8]: need blank line in the end or not?

File: docs/source/installation/grass_cluster_provisioning_on_premises.rst [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+=====================================================
+
+With the following guide, you can build up a MARO cluster in
+`grass mode <../distributed_training/orchestration_with_grass.html#orchestration-with-grass>`_
+in local private network and run your training job in On-Premises distributed environment.
+
+Prerequisites
+-------------
+
+* Liunx with Python 3.6+
```
Comment (chaosddp) [inline docs/source/installation/grass_cluster_provisioning_on_premises.rst:12]: Linux

File: docs/source/installation/multi_processes_localhost_provisioning.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
+Multi-processes Localhost Provisioning
+=========================================
+With the following guide, it is easy to implement your training jobs through
+the multi-processes in the localhost environment.
+
+Prerequisites
+-------------
+* Liunx with Python 3.6+
```
Comment (chaosddp) [inline docs/source/installation/multi_processes_localhost_provisioning.rst:8]: Linux

**Review by Jinyu-W — DISMISSED** [review]

**Review by chaosddp — COMMENTED** [review]
2 typo

**Review by chaosddp — APPROVED** [review]

**Pull request merged**